pmd/pmd.xml

```diff
@@ -8,46 +8,46 @@
-  <rule ref="category/java/multithreading.xml/DoubleCheckedLocking" deprecated="true" />
+  <rule ref="category/java/multithreading.xml/AvoidThreadGroup" />
+  <rule ref="category/java/multithreading.xml/DontCallThreadRun" />
+  <rule ref="category/java/multithreading.xml/DoubleCheckedLocking" />
 
-  <rule ref="category/java/bestpractices.xml/AvoidUsingHardCodedIP" deprecated="true" />
-  <rule ref="category/java/bestpractices.xml/CheckResultSet" deprecated="true" />
+  <rule ref="category/java/bestpractices.xml/AvoidUsingHardCodedIP" />
+  <rule ref="category/java/bestpractices.xml/CheckResultSet" />
 
-  <rule ref="category/java/codestyle.xml/ExtendsObject" deprecated="true" />
-  <rule ref="category/java/codestyle.xml/ForLoopShouldBeWhileLoop" deprecated="true" />
+  <rule ref="category/java/codestyle.xml/ExtendsObject" />
+  <rule ref="category/java/codestyle.xml/ForLoopShouldBeWhileLoop" />
 
-  <rule ref="category/java/performance.xml/BigIntegerInstantiation" deprecated="true" />
-  <rule ref="category/java/performance.xml/BooleanInstantiation" deprecated="true" />
+  <rule ref="category/java/performance.xml/BigIntegerInstantiation" />
+  <rule ref="category/java/performance.xml/BooleanInstantiation" />
 
-  <rule ref="category/java/design.xml/CollapsibleIfStatements" deprecated="true" />
-  <rule ref="category/java/design.xml/SimplifiedTernary" deprecated="true" />
+  <rule ref="category/java/design.xml/CollapsibleIfStatements" />
+  <rule ref="category/java/design.xml/SimplifiedTernary" />
 <!--  inline <rule ref="rulesets/java/braces.xml"/>-->
-  <rule ref="category/java/codestyle.xml/ForLoopsMustUseBraces" deprecated="true" />
-  <rule ref="category/java/codestyle.xml/IfElseStmtsMustUseBraces" deprecated="true" />
-  <rule ref="category/java/codestyle.xml/IfStmtsMustUseBraces" deprecated="true" />
-  <rule ref="category/java/codestyle.xml/WhileLoopsMustUseBraces" deprecated="true" />
-  <rule ref="rulesets/java/clone.xml"/>
-  <rule ref="rulesets/java/finalizers.xml"/>
-  <rule ref="rulesets/java/logging-java.xml">
-      <exclude name="GuardLogStatementJavaUtil"/>
-  </rule>
-  <rule ref="rulesets/java/migrating.xml"/>
-  <rule ref="rulesets/java/sunsecure.xml"/>
-  <rule ref="rulesets/java/unusedcode.xml"/>
+  <rule ref="category/java/codestyle.xml/ForLoopsMustUseBraces" />
+  <rule ref="category/java/codestyle.xml/IfElseStmtsMustUseBraces" />
+  <rule ref="category/java/codestyle.xml/IfStmtsMustUseBraces" />
+  <rule ref="category/java/codestyle.xml/WhileLoopsMustUseBraces" />
+<!--  <rule ref="rulesets/java/clone.xml"/>-->
+<!--  <rule ref="rulesets/java/finalizers.xml"/>-->
+<!--  <rule ref="rulesets/java/logging-java.xml">-->
+<!--      <exclude name="GuardLogStatementJavaUtil"/>-->
+<!--  </rule>-->
+<!--  <rule ref="rulesets/java/migrating.xml"/>-->
+<!--  <rule ref="rulesets/java/sunsecure.xml"/>-->
+<!--  <rule ref="rulesets/java/unusedcode.xml"/>-->
```

Comment:
why are these commented out now?

Reply:
they are deprecated, and contained many other rules which i thought can be added in a second step, as nobody cared for these rules for so long:

https://pmd.github.io/pmd/pmd_rules_java.html